--- a/local/Makefile.in	2010-09-01 15:03:23.000000000 -0700
+++ b/local/Makefile.in	2010-09-01 15:04:10.976111119 -0700
@@ -101,7 +101,7 @@ tkmib.made: $(srcdir)/tkmib
 
 mib2c.made: $(srcdir)/mib2c
 	if test "x$(PERL)" != "x" ; then \
-	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;' ${srcdir}/mib2c > mib2c.made; \
+	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;s#/usr/local/etc/snmp#$(SNMPCONFPATH)#;' ${srcdir}/mib2c > mib2c.made; \
 	else \
 	  touch mib2c.made; \
         fi
--- a/mibs/Makefile.in	2010-09-01 15:03:23.000000000 -0700
+++ b/mibs/Makefile.in	2010-09-01 15:04:10.976111119 -0700
@@ -43,16 +43,22 @@ RFCMIBS	= IF-MIB.txt IF-INVERTED-STACK-M
 	SCTP-MIB.txt
 
 NETSNMPMIBS = NET-SNMP-TC.txt NET-SNMP-MIB.txt NET-SNMP-AGENT-MIB.txt \
-	NET-SNMP-EXAMPLES-MIB.txt NET-SNMP-EXTEND-MIB.txt NET-SNMP-PASS-MIB.txt
+	NET-SNMP-EXAMPLES-MIB.txt NET-SNMP-EXTEND-MIB.txt NET-SNMP-PASS-MIB.txt \
+	NET-SNMP-MONITOR-MIB.txt NET-SNMP-SYSTEM-MIB.txt \
+	NET-SNMP-VACM-MIB.txt
 
 UCDMIBS = UCD-SNMP-MIB.txt UCD-DEMO-MIB.txt UCD-IPFWACC-MIB.txt \
 	UCD-DLMOD-MIB.txt UCD-DISKIO-MIB.txt
 
+EXTRAMIBS = BGP4-MIB.txt BRIDGE-MIB.txt OSPF-MIB.txt \
+	OSPF-TRAP-MIB.txt RIPv2-MIB.txt SOURCE-ROUTING-MIB.txt \
+	LM-SENSORS-MIB.txt
+
 DEFAULTMIBS = @default_mibs_install@
 
 MIBS	= $(V1MIBS) $(V2MIBS) $(V3MIBS) $(RFCMIBS) \
 	$(AGENTMIBS) $(IANAMIBS) \
-	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS)
+	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS) $(EXTRAMIBS)
 
 all: standardall
 
