--- net-snmp-5.3.1/apps/snmptrapd.c.orig	2007-07-06 17:02:00.000000000 +0200
+++ net-snmp-5.3.1/apps/snmptrapd.c	2007-07-31 16:45:38.000000000 +0200
@@ -596,6 +596,16 @@
 #endif
     netsnmp_trapd_handler *traph;
 
+#ifndef WIN32
+    /*
+     * close all non-standard file descriptors we may have
+     * inherited from the shell.
+     */
+    for (i = getdtablesize() - 1; i > 2; --i) {
+	(void) close(i);
+    }
+#endif /* #WIN32 */
+
 #ifdef SIGTERM
     signal(SIGTERM, term_handler);
 #endif
