--- net-snmp-5.4.1.orig/snmplib/snmp_logging.c	2006/12/04 17:54:15	15632
+++ net-snmp-5.4.1/snmplib/snmp_logging.c	2007/10/14 19:07:24	16721
@@ -179,6 +179,9 @@
 {
     int pri_low = LOG_DEBUG;
 
+    if (optarg == NULL)
+        return -1;
+
     switch (*optarg) {
         case '0': 
         case '!': 
@@ -384,7 +387,7 @@
     case 'S':
         priority = decode_priority( optarg, &pri_max );
         if (priority == -1 || !argv)  return -1;
-        optarg = argv[++optind];
+        optarg++;
         /* Fallthrough */
     case 's':
         if (inc_optind)
