--- net-snmp-5.4.1~dfsg.orig/configure.in
+++ net-snmp-5.4.1~dfsg/configure.in
@@ -2854,6 +2854,9 @@
 AC_CHECK_FUNC(kvm_getswapinfo, AC_DEFINE(HAVE_KVM_GETSWAPINFO, 1, [Define to 1 if you have the `kvm_getswapinfo' function.]),
 	AC_CHECK_LIB(kvm, kvm_getswapinfo, AC_DEFINE(HAVE_KVM_GETSWAPINFO, 1, [Define to 1 if you have the `kvm_getswapinfo' function.])
 	_libkvm="-lkvm"))
+AC_CHECK_FUNC(nlist, AC_DEFINE(HAVE_NLIST, 1, [Define to 1 if you have the `nlist' function.]),
+        AC_CHECK_LIB(kvm, nlist, AC_DEFINE(HAVE_NLIST, 1, [Define to 1 if you have the `nlist' function.])
+        _libkvm="-lkvm"))
 if test "x${_libkvm}" != "x"; then
 	AC_DEFINE(HAVE_LIBKVM, 1, [Define to 1 if you have the `kvm' library (-lkvm).])
 	LAGENTLIBS="$LAGENTLIBS ${_libkvm}"
