Description: fix typo
Author: Hideki Yamane <henrich@debian.org>
---
Origin: vendor
Bug-Debian: http://bugs.debian.org/603593
Forwarded: no
Last-Update: 2013-01-23

--- net-snmp-5.7.2~dfsg.orig/EXAMPLE.conf.def
+++ net-snmp-5.7.2~dfsg/EXAMPLE.conf.def
@@ -63,7 +63,7 @@ view   systemonly  included   .1.3.6.1.2
 #rwuser   authPrivUser   priv
 
 #  It's no longer typically necessary to use the full 'com2sec/group/access' configuration
-#  r[ou]user and r[ow]community, together with suitable views, should cover most requirements
+#  r[ow]user and r[ow]community, together with suitable views, should cover most requirements
 
 
 
