Description: fix spelling error
Author: Hideki Yamane <henrich@debian.org>

---
Origin: vendor
Forwarded: no
Last-Update: 2013-01-22

--- net-snmp-5.7.2~dfsg.orig/apps/encode_keychange.c
+++ net-snmp-5.7.2~dfsg/apps/encode_keychange.c
@@ -403,7 +403,7 @@ usage_to_file(FILE * ofp)
    "-f will require reading from the stdin/terminal, ignoring a) and b).\n\
     -P will prevent prompts for passphrases to stdout from being printed.\n\
 \n\
-    <engineID> is interpreted as a hex string when preceeded by \"0x\",\n\
+    <engineID> is interpreted as a hex string when preceded by \"0x\",\n\
     otherwise it is created to contain \"text\".  If nothing is given,\n\
     <engineID> is constructed from the first IP address for the local host.\n");
 
--- net-snmp-5.7.2~dfsg.orig/apps/snmpusm.c
+++ net-snmp-5.7.2~dfsg/apps/snmpusm.c
@@ -150,7 +150,7 @@ usage(void)
     fprintf(stderr, "\t\t\t(it won't be active until you active it)\n");
     fprintf(stderr, "\t-Cx\t\tChange the privacy key.\n");
     fprintf(stderr, "\t-Ca\t\tChange the authentication key.\n");
-    fprintf(stderr, "\t-Ck\t\tAllows to use localized key (must start with 0x)\n");
+    fprintf(stderr, "\t-Ck\t\tAllows one to use localized key (must start with 0x)\n");
     fprintf(stderr, "\t\t\tinstead of passphrase.\n");
 }
 
--- net-snmp-5.7.2~dfsg.orig/apps/snmpnetstat/inet6.c
+++ net-snmp-5.7.2~dfsg/apps/snmpnetstat/inet6.c
@@ -333,7 +333,7 @@ icmp6_stats(const char *name)
     struct stat_table icmp6stats_tbl[] = {
         {1, "%d total message%s received"},
         {2, "%d message%s dropped due to errors"},
-        {18, "%d ouput message request%s"},
+        {18, "%d output message request%s"},
         {19, "%d output message%s discarded"},
         {0, ""}
     };
--- net-snmp-5.7.2~dfsg.orig/apps/snmpnetstat/inet.c
+++ net-snmp-5.7.2~dfsg/apps/snmpnetstat/inet.c
@@ -445,7 +445,7 @@ icmp_stats(const char *name)
     struct stat_table icmpstats_tbl[] = {
         {1, "%d total message%s received"},
         {2, "%d message%s dropped due to errors"},
-        {14, "%d ouput message request%s"},
+        {14, "%d output message request%s"},
         {15, "%d output message%s discarded"},
         {0, ""}
     };
