Description: fix man description
Author: Hideki Yamane <henrich@debian.org>

---
Origin: vendor
Bug-Debian: http://bugs.debian.org/616437
Forwarded: no
Last-Update: 2013-01-23

--- net-snmp-5.7.2~dfsg.orig/man/snmpd.8.def
+++ net-snmp-5.7.2~dfsg/man/snmpd.8.def
@@ -78,9 +78,9 @@ To get a list of compiled modules, run t
 .I "\-Dmib_init \-H"
 (assuming debugging support has been compiled in).
 .TP
-.B \-L[efos]
+.B \-L[eEfFoOsS]
 Specify where logging output should be directed (standard error or output,
-to a file or via syslog).  See LOGGING OPTIONS in snmpcmd(5) for details.
+to a file or via syslog).  See LOGGING OPTIONS in snmpcmd(1) for details.
 .TP
 .BR \-m " \fIMIBLIST"
 Specifies a colon separated list of MIB modules to load for this
